hnswlib/bruteforce.h

```diff
@@ -92,20 +92,24 @@ namespace hnswlib {
 
 
         std::priority_queue<std::pair<dist_t, labeltype >>
-        searchKnn(const void *query_data, size_t k) const {
+        searchKnn(const void *query_data, size_t k, filter_func_t isIdAllowed=allowAllIds) const {
```

Comment:
Would not it better to pass NULL as a default value to show that we do not do filtering by default (see https://github.com/nmslib/hnswlib/pull/402/commits/ad3440c83555d9a76eef0e23bc6505c86b026716#r954851933)?

`filter_func_t isIdAllowed=NULL`